chanfitness/chaneventstore_test.go

```diff
@@ -466,3 +472,59 @@ func TestGetUptime(t *testing.T) {
 		})
 	}
 }
+
+// TestAddChannel tests that channels are added to the event store with
+// appropriate timestamps regardless of whether the peer is online. This test
+// addresses a bug where offline channels did not have an opened time set.
+func TestAddChannel(t *testing.T) {
+	_, vertex, chanPoint := getTestChannel(t)
+
+	tests := []struct {
+		name          string
+		peerOnline    bool
+		expctedEvents []*channelEvent
+	}{
+		{
+			name:       "peer online on start",
+			peerOnline: true,
+			expctedEvents: []*channelEvent{{
+				eventType: peerOnlineEvent,
+			}},
+		},
+		{
+			name:          "peer offline on start",
+			peerOnline:    true,
+			expctedEvents: []*channelEvent{},
+		},
+	}
+
+	for _, test := range tests {
+		test := test
+
+		t.Run(test.name, func(t *testing.T) {
+			store := NewChannelEventStore(&Config{})
+
+			// Add channel to the store.
+			store.addChannel(chanPoint, vertex, test.peerOnline)
+
+			// Check that the eventlog is successfully added.
+			eventlog, ok := store.channels[chanPoint]
+			if !ok {
+				t.Fatalf("channel should be in store")
+			}
+
+			// Ensure that open time is always set.
+			if eventlog.openedAt.IsZero() {
+				t.Fatalf("channel should have opened at set")
+			}
+
+			// Check that an initial event has been added, where appropriate.
+			for i, event := range test.expctedEvents {
+				if eventlog.events[i].eventType != event.eventType {
+					t.Fatalf("expected: %v, got: %v", event.eventType,
+						eventlog.events[i].eventType)
+				}
+			}
```

Comment:
No uptime assertion, which was what the bug was about?

Reply:
The bug was that we had a zero open time for channels that started with offline peers, which is checked above. The effect of a zero open time was that lifespan would be calculated as `endtime -0` instead of `endtime - openedAt`.

Reply:
Ah right. I was thinking about the tiny gap between init and the first event, but probably hardly measurable